pandas/tseries/frequencies.py

```diff
@@ -213,6 +211,18 @@ def __init__(self, index, warn: bool = True) -> None:
         self.index = index
         self.i8values = index.asi8
 
+        # For get_unit_from_dtype we need the dtype to the underlying ndarray,
+        #  which for tz-aware is not the same as index.dtype
+        if isinstance(index, ABCIndex):
+            # error: Item "ndarray[Any, Any]" of "Union[ExtensionArray,
+            # ndarray[Any, Any]]" has no attribute "_ndarray"
+            self._reso = get_unit_from_dtype(
```

Comment:
Is the DTI/TDI cases already tested? I saw the DTA/TDA cases you added

Reply:
Yes